docs/contributor/arch/001-otel-collector-metric-based-pipeline-status.md

```diff
@@ -0,0 +1,102 @@
+# 1. Trace/Metric Pipeline status based on OTel Collector metrics
+
+Date: 2023-08-11
+
+## Status
+
+Proposed
+
+## Context
```

Comment:
Let's link https://github.com/kyma-project/telemetry-manager/issues/425 here. The individual section feel a bit like loose ends without context (which is okay, since this is an ADR and not a full design document).